### PR TITLE
Misc Tasks

### PR DESCRIPTION
**Pull request by StefanGreve**
Miscellaneous small tasks that require attention.

## TODO

- [x] Add linter through Git hooks
- [x] Add branch protection rules for master
- [x] Add custom favicon
- [x] edit meta tags
- [x] rewrite readme file
- [x] think about other things to add in preparation for deployment
- [x] embedded [this icon library](https://github.com/ng-icons/ng-icons) and remove all references to `adv-icons`

### CODE REVIEW

File: src/assets/favicon/dark-apple-touch-icon.png
Comment (KidBased) [inline src/assets/favicon/dark-apple-touch-icon.png:1]: Looks smooth.

File: src/app/directives/clicked-outside.directive.ts
Comment (KidBased) [inline src/app/directives/clicked-outside.directive.ts:1]: Hm, I don't quite understand why you used the `never` type here. Mind giving me a short explanation?

File: src/app/app.component.ts [hunk 1]
```diff
@@ -28,30 +30,36 @@ export class AppComponent {
 
   @ViewChild(DialogComponent, {static: false})
   settingsDialog: DialogComponent | undefined;
+
   settingsSubject$: BehaviorSubject<boolean> = new BehaviorSubject(false);
 
+  ngOnInit(): void {
+    this.isDarkThemeEnabled = this.themeService.getActiveTheme === Theme.Dark;
```
Comment (StefanGreve) [inline src/app/app.component.ts:37]: I think the implementation of the `getActiveTheme` function should return the `getPreferredBrowserTheme` if there's no theme preference stored in local storage instead of the fallback default (light). Thoughts?

Comment (KidBased) [reply src/app/app.component.ts:37]: I agree with that.

File: src/app/directives/clicked-outside.directive.ts
Comment (StefanGreve) [reply src/app/directives/clicked-outside.directive.ts:1]: I just reviewed the definition of this keyword, and from what I learnt about it `never` seems to be misplaced here. However, `any` should be avoided if not needed, do you have an alternative suggestions? I wasn't sure if there could be inferred a non-generic type here

File: src/app/services/theme-switcher.service.ts [hunk 2]
```diff
@@ -10,9 +10,11 @@ import { ThemeDefinition } from "../interfaces/theme";
 export class ThemeSwitcherService {
   public readonly localStorageKey = "theme";
   private readonly fallbackTheme = "Light";
+  private preferredBrowserTheme = "";
 
   constructor() {
-    const cachedTheme = localStorage.getItem(this.localStorageKey) || this.fallbackTheme;
+    this.preferredBrowserTheme = this.getPreferredBrowserTheme === 1 ? "Light" : "Dark";
```
Comment (StefanGreve) [inline src/app/services/theme-switcher.service.ts:16]: I think this line can be simplified to `const cachedTheme = Theme[this.getPreferredBrowserTheme];`. This should return the enum name as string, perhaps you need to chain an additional `toString()` afterwards.

**Comment by StefanGreve** [comment]
@KidBased Remember to use simple present in your commit messages, not simple past.

**Comment by KidBased** [comment]
> @KidBased Remember to use simple present in your commit messages, not simple past.

Thx, I completely forgot about that.

**Review by KidBased — DISMISSED** [review]

File: src/app/services/theme-switcher.service.ts [hunk 3]
```diff
@@ -54,7 +52,7 @@ export class ThemeSwitcherService {
   }
 
   get getActiveTheme(): Theme {
-    const activeTheme: string = localStorage.getItem(this.localStorageKey) || this.preferredBrowserTheme;
```
Comment (StefanGreve) [inline src/app/services/theme-switcher.service.ts]: Why did you remove the local storage access? This line should be

```ts
const activeTheme: string = localStorage.getItem(this.localStorageKey) || Theme[this.getPreferredBrowserTheme];
```

User-configured app settings should be persistent, after all.

**Review by StefanGreve — COMMENTED** [review]
@KidBased One small thing requires your attention, everything else is fine.

File: src/app/services/theme-switcher.service.ts (hunk 3 above)
Comment (StefanGreve) [reply src/app/services/theme-switcher.service.ts]: This change actually resulted in an error, which is why I fixed it myself.

**Review by KidBased — APPROVED** [review]
